flow-report/src/help-dialog.tsx

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {FunctionComponent, JSX} from 'preact';
+
+import {useUIStrings} from './i18n/i18n';
+import {CloseIcon, NavigationIcon, SnapshotIcon, TimespanIcon} from './icons';
+import {classNames} from './util';
+
+const HelpDialogColumn: FunctionComponent<{
+  icon: JSX.Element,
+  userFriendlyModeLabel: string;
+  lighthouseOfficialModeLabel: string;
+  modeDescription: string;
+  useCaseInstruction: string;
+  useCases: string[];
+  categoriesLabel: string;
+  availableCategories: string[];
+}> = (props) => {
+  return (
+    <div className="HelpDialogColumn">
+      <div className="HelpDialogColumn__legend">
+        <div className="HelpDialogColumnTimeline">
+          {props.icon}
+          <div className="HelpDialogColumnTimeline__line" />
+        </div>
+        <div className="HelpDialogColumn__legend-label">{props.userFriendlyModeLabel}</div>
+      </div>
+      <div className="HelpDialogColumn__header">
+        <div className="HelpDialogColumn__header-title">{props.lighthouseOfficialModeLabel}</div>
+        <p>{props.modeDescription}</p>
+      </div>
+      <div className="HelpDialogColumn__use-cases">
+        <p>{props.useCaseInstruction}</p>
+        <ul>
+          {props.useCases.map((useCase, i) => (
+            <li key={i}>{useCase}</li>
+          ))}
+        </ul>
+      </div>
+      <div className="HelpDialogColumn__categories">
+        <p>{props.categoriesLabel}</p>
+        <ul>
+          {props.availableCategories.map((category, i) => (
+            <li key={i}>{category}</li>
+          ))}
+        </ul>
+      </div>
+    </div>
+  );
+};
+
+export const HelpDialog: FunctionComponent<{visible: boolean; onClose: () => void}> = ({
+  visible,
+  onClose,
+}) => {
+  const strings = useUIStrings();
+
+  return (
+    <div className={classNames('HelpDialog', {'HelpDialog--hidden': !visible})}>
```

Comment:
Instead of adding a `--hidden` style, WDYT about removing the node entirely? This way you don't need the `visible` param and it won't be rendered in the DOM for most use cases.

Reply:
sure we can go with conditional rendering here. my thoughts on this topic in general are to default to `display: none` if the content is not exceedingly expensive to keep around in the DOM because you get DOM-based state preservation for free, and it avoids really thorny bugs in many SSR hydration implementations. in this case though, we're definitely not SSR'ing and there's no state to keep, so I'm relatively indifferent.